src/Helper/Service/IdentityFactory.php

```diff
@@ -25,10 +26,22 @@ class IdentityFactory implements FactoryInterface
      */
     public function __invoke(ContainerInterface $container, $name, array $options = null)
     {
+        $container = $container->getServiceLocator();
```

Comment:
Thanks for the tip!

Yes, I'd like to make it work with both v2 and v3, so this will be helpful.